### DISCUSSION TITLE
how to connect other substrate chains?

### DISCUSSION THREAD

**Discussion by Yxiaoyang** [polkascan/py-substrate-interface#303] [Q&A] [answered]
so sorry，i have a question, i want connect a substrate chain, but I only have the official types.json, i want know can it work in python？ or tell me, what should I do

please, thanks for much.

**Comment by arjanz** [comment, marked as answer]
Basically you will have to tell the libraries how to interprete the types present in the runtime. 

Some pointers to help you:

1. Does your Substrate chain support `MetadataV14`? Then you need to upgrade this library to version `1.0.x` and the types will be included and will most likely work out of the box
2. With a runtime using `MetadataV13` or less you need to specify the types yourself in a certain format. See https://github.com/polkascan/py-substrate-interface#manually-set-required-properties. Determine your starting point, this will be `type_registry_preset=development` for vanilla Substrate. See https://github.com/polkascan/py-substrate-interface/issues/122#issuecomment-874585356 for more info.
3. If you have a types.json you use for PolkadotJS, you will have to reformat that into this format: https://github.com/polkascan/py-scale-codec/blob/master/scalecodec/type_registry/test.json Another example, for Acala I converted the PolkadotJs types.json some time ago into https://gist.github.com/arjanz/b7290f68d5d94b83e7dd4e20e4f6ef3c

Hope this helps!

**Comment by Yxiaoyang** [comment]
hello，excuse me，do you have doc to reformat the "type.json"? or Can I provide you with the type.json file, can you help?

**Comment by Yxiaoyang** [comment]
sorry, i'm ok. thanks for much
